Competition/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java

```diff
@@ -0,0 +1,133 @@
+package competition.subsystems.elevator;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.log4j.Logger;
+
+import competition.ElectricalContract2019;
+import edu.wpi.first.wpilibj.PIDController;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.controls.sensors.XDigitalInput;
+import xbot.common.injection.wpi_factories.CommonLibFactory;
+import xbot.common.math.MathUtils;
+import xbot.common.math.PID;
+import xbot.common.math.PIDFactory;
+import xbot.common.properties.XPropertyManager;
+
+@Singleton
+public class ElevatorSubsystem extends BaseSubsystem {
+    private static Logger log = Logger.getLogger(ElevatorSubsystem.class);
+
+    public XCANTalon master;
+    public XDigitalInput calibrationSensor;
+    private double calibrationSensorPosition = -1;
+    private boolean isCalibrated;
+    private double power;
+
+    public enum HatchLevel {
+        Low, Medium, High,
+    }
+
+    @Inject
+    public ElevatorSubsystem(CommonLibFactory factory, XPropertyManager propManager, ElectricalContract2019 contract,
+            PIDFactory pd) {
+        log.info("Creating ElevatorSubsystem");
+        this.master = factory.createCANTalon(contract.getElevatorMasterMotor().channel);
+        master.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 0);
+        this.calibrationSensor = factory.createDigitalInput(contract.getElevatorCalibrationSensor().channel);
+    }
+
+    public void stop() {
+        setPower(0);
+    }
+
+    public void raiseElevator() {
+        setPower(1);
+    }
+
+    public void lowerElevator() {
+        setPower(-1);
+    }
+
+    // Set's isCalibrated to true is sensor button is pressed
+    public boolean isCalibrationSensorPressed() {
+        return calibrationSensor.get();
+    }
+
+    public boolean calibrationSensorState() {
+        if (isCalibrationSensorPressed()) {
+            isCalibrated = true;
+        }
+        return isCalibrated;
+    }
+
+    public void lowerElevatorToCalibrationPoint() {
```

Comment:
I don't think we need this, now that your elevator automatically stops when it reaches the lower point.